src/components/cylc/tree/cylc-tree.js

```diff
@@ -230,7 +230,15 @@ class CylcTree {
   addCyclePoint (cyclePoint) {
     if (!this.lookup.has(cyclePoint.id)) {
       this.lookup.set(cyclePoint.id, cyclePoint)
-      this.root.children.push(cyclePoint)
+      const parent = this.root
+      const names = this.root.children
+        .map(child => child.node.name)
+        .reverse()
+      const insertIndex = sortedIndex(names, cyclePoint.node.name)
+      // cycle points are inserted in the reverse order, so we have to handle that sortedIndex will give you the
+      // wrong index (except when the list is empty). That's why we reverse the list first, to get the index, and
+      // find where it would be in the not-reversed list.
+      parent.children.splice(parent.children.length - insertIndex, 0, cyclePoint)
```

Comment:
In the case where the index is `0`, it won't be negative. Shouldn't matter that we used `.splice(0, 0, elem)` instead of simply `.push(elem)` I believe.